src/Common/Commands.Common/CustomAttributes/BreakingChangeAttributeHelper.cs

```diff
@@ -0,0 +1,151 @@
+
+    https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes/breaking-change-template.md
+    -->";
+
+        public static void processCustomAttributesAtRuntime(Type type, List<String> boundParameterNames)
+        {
+            bool emitWarningOrError = true;
+
+            try
+            {
+                emitWarningOrError = Boolean.Parse(System.Environment.GetEnvironmentVariable(SUPPRESS_ERROR_OR_WARNING_MESSAGE_ENV_VARIABLE_NAME));
+            }
+            catch (Exception)
+            {
+                //no action
+            }
+
+            if (!emitWarningOrError)
+            {
+                //Do not process the attributes at runtime... The env variable to override the warning messages is set
+                return;
+            }
+
+            List<GenericBreakingChangeAttribute> attributes = getAllBreakingChangeAttributesInType(type, boundParameterNames);
+
+            if (attributes != null && attributes.Count > 0)
+            {
+                Console.WriteLine("Breaking changes in the cmdlet : " + Utilities.getNameFromCmdletType(type) + "\n");
+            }
+
+            foreach (GenericBreakingChangeAttribute attribute in attributes)
+            {
+                attribute.printCustomAttributeInfo(type, false);
+            }
+        }
+
+        public static List<String> getBreakingChangeMessagesForType(Type type)
+        {
+            List<String> messages = new List<string>();
+
+            //This is used as a migration guide, we need to process all properties/fields, moreover at this point of time we do not have a list of all the
+            //bound params anyways
+            foreach (GenericBreakingChangeAttribute attribute in getAllBreakingChangeAttributesInType(type, null))
+            {
+                messages.Add(attribute.getBreakingChangeTextFromAttribute(type, true));
+            }
+
+            return messages;
+        }
+
+        public static List<GenericBreakingChangeAttribute> getAllBreakingChangeAttributesInType(Type type, List<String> boundParameterNames)
+        {
+            List<GenericBreakingChangeAttribute> attributeList = new List<GenericBreakingChangeAttribute>();
+
+            attributeList.AddRange(type.GetCustomAttributes(typeof(GenericBreakingChangeAttribute), false).Cast<GenericBreakingChangeAttribute>());
+
+            foreach (MethodInfo m in type.GetRuntimeMethods())
+            {
+                attributeList.AddRange(m.GetCustomAttributes(typeof(GenericBreakingChangeAttribute), false).Cast<GenericBreakingChangeAttribute>());
+            }
+
+            //Only process the fields if the bound params have the field Name in them. If the bound params is null then
+            //we do no filtering
+            foreach (FieldInfo f in type.GetRuntimeFields())
+            {
+                if ((boundParameterNames != null && boundParameterNames.Contains(f.Name)) || (boundParameterNames == null))
```

Comment:
Rather than putting this logic in the cmdlet, seems like the attribute itself should be able to detect whether it should be applied or not - perhaps the attribute shou;d have a ```TestInvocation``` or similar method that takes a CmdletInvocation and returns 'true' if the breaking change applies to this invocation.  Or, the parameter attributes could have a ```TestRuntimeParameter``` method that takes a BoundParameters dictionary and returns true or false based on whether the parameter was invoked as intended.  I prefer the ```TestInvocation``` a bit better, because it provides more future flexibility

Reply:
done